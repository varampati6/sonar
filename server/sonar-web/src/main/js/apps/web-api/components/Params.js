/*
 * SonarQube
 * Copyright (C) 2009-2017 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
// @flow
import React from 'react';
import InternalBadge from './InternalBadge';
import DeprecatedBadge from './DeprecatedBadge';
import type { Param } from '../../../api/web-api';

export default class Params extends React.PureComponent {
  props: {
    showDeprecated: boolean,
    showInternal: boolean,
    params: Array<Param>
  };

  render() {
    const { showDeprecated, showInternal, params } = this.props;
    const displayedParameters = params
      .filter(p => showDeprecated || !p.deprecatedSince)
      .filter(p => showInternal || !p.internal);
    return (
      <div className="web-api-params">
        <table>
          <tbody>
            {displayedParameters.map(param => (
              <tr key={param.key}>
                <td className="markdown" style={{ width: 180 }}>
                  <code>{param.key}</code>

                  {param.internal &&
                    <div className="little-spacer-top">
                      <InternalBadge />
                    </div>}

                  {param.deprecatedSince &&
                    <div className="little-spacer-top">
                      <DeprecatedBadge since={param.deprecatedSince} />
                    </div>}

                  {showDeprecated &&
                    param.deprecatedKey &&
                    <div className="little-spacer-top">
                      <code>{param.deprecatedKey}</code>
                    </div>}

                  {showDeprecated &&
                    param.deprecatedKey &&
                    param.deprecatedKeySince &&
                    <div className="little-spacer-top">
                      <DeprecatedBadge since={param.deprecatedKeySince} />
                    </div>}

                  <div className="note little-spacer-top">
                    {param.required ? 'required' : 'optional'}
                  </div>

                  {param.since &&
                    <div className="note little-spacer-top">
                      since {param.since}
                    </div>}
                </td>

                <td>
                  <div
                    className="markdown"
                    dangerouslySetInnerHTML={{ __html: param.description }}
                  />
                </td>

                <td style={{ width: 250 }}>
                  {param.possibleValues &&
                    <div>
                      <h4>Possible values</h4>
                      <ul className="list-styled">
                        {param.possibleValues.map(value => (
                          <li key={value} className="little-spacer-top">
                            <code>{value}</code>
                          </li>
                        ))}
                      </ul>
                    </div>}

                  {param.defaultValue &&
                    <div className="little-spacer-top">
                      <h4>Default value</h4>
                      <code>{param.defaultValue}</code>
                    </div>}

                  {param.exampleValue &&
                    <div className="little-spacer-top">
                      <h4>Example value</h4>
                      <code>{param.exampleValue}</code>
                    </div>}

                  {param.maxValuesAllowed != null &&
                    <div className="little-spacer-top">
                      <h4>Maximum allowed values</h4>
                      <code>{param.maxValuesAllowed}</code>
                    </div>}
                </td>
              </tr>
            ))}
          </tbody>
        </table>
      </div>
    );
  }
}
