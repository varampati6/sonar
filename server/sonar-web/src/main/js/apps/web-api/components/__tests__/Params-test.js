/*
 * SonarQube
 * Copyright (C) 2009-2017 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import React from 'react';
import { shallow } from 'enzyme';
import Params from '../Params';

it('should render deprecated parameters', () => {
  const params = [
    {
      key: 'foo',
      deprecatedSince: '5.0'
    }
  ];
  expect(shallow(<Params params={params} showDeprecated={true} />)).toMatchSnapshot();
});

it('should not render deprecated parameters', () => {
  const params = [
    {
      key: 'foo',
      deprecatedSince: '5.0'
    }
  ];
  expect(shallow(<Params params={params} showDeprecated={false} />)).toMatchSnapshot();
});

it('should render deprecated key', () => {
  const params = [
    {
      key: 'foo',
      deprecatedKey: 'foo-deprecated',
      deprecatedKeySince: '5.0'
    }
  ];
  expect(shallow(<Params params={params} showDeprecated={true} />)).toMatchSnapshot();
});
