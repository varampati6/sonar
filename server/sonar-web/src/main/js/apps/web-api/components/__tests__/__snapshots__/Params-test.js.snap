// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not render deprecated parameters 1`] = `
<div
  className="web-api-params"
>
  <table>
    <tbody />
  </table>
</div>
`;

exports[`should render deprecated key 1`] = `
<div
  className="web-api-params"
>
  <table>
    <tbody>
      <tr>
        <td
          className="markdown"
          style={
            Object {
              "width": 180,
            }
          }
        >
          <code>
            foo
          </code>
          <div
            className="little-spacer-top"
          >
            <code>
              foo-deprecated
            </code>
          </div>
          <div
            className="little-spacer-top"
          >
            <DeprecatedBadge
              since="5.0"
            />
          </div>
          <div
            className="note little-spacer-top"
          >
            optional
          </div>
        </td>
        <td>
          <div
            className="markdown"
            dangerouslySetInnerHTML={
              Object {
                "__html": undefined,
              }
            }
          />
        </td>
        <td
          style={
            Object {
              "width": 250,
            }
          }
        />
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`should render deprecated parameters 1`] = `
<div
  className="web-api-params"
>
  <table>
    <tbody>
      <tr>
        <td
          className="markdown"
          style={
            Object {
              "width": 180,
            }
          }
        >
          <code>
            foo
          </code>
          <div
            className="little-spacer-top"
          >
            <DeprecatedBadge
              since="5.0"
            />
          </div>
          <div
            className="note little-spacer-top"
          >
            optional
          </div>
        </td>
        <td>
          <div
            className="markdown"
            dangerouslySetInnerHTML={
              Object {
                "__html": undefined,
              }
            }
          />
        </td>
        <td
          style={
            Object {
              "width": 250,
            }
          }
        />
      </tr>
    </tbody>
  </table>
</div>
`;
